extensions/mongoProcessor/MongoQueueProcessor.js

```diff
@@ -273,18 +276,33 @@ class MongoQueueProcessor {
      */
     _updateLocations(entry, location) {
         const locations = entry.getLocation();
-        const editLocations = locations.map(l => {
-            const newValues = {
+        if (!locations || locations.length === 0) {
+            const dataStoreVersionId = entry.getVersionId() ?
```

Comment:
Can't this be accessed with `entry.getDataStoreVersionId()`? That way we'd preserve the public cloud's version ID that's stored here, if I'm not mistaken.

Reply:
For 0-byte objects and delete markers, the locations field will be empty. 
For non-0-byte objects, the locations field don't have dataStoreVersionId defined. I'm thinking since the objects I'm testing are created on s3c, it does not define this field itself, so we can't rely on this field to be set when ingesting these objects.

Do you know cases where this field will be set? I can only think of Zenko use-cases, but not sure of cases for s3c

Reply:
It looks like the `dataStoreVersionId` field is used in versions of S3C going back at least to 7.4.2. I'm not sure whether that affects this feature though.

Reply:
My understanding is that if S3C is using the multiple backend, you may end up with locations storing the value. @rahulreddy would probably have a better sense though!

Reply:
S3C doesn't support any of the cloud backends even if the code is in there. So it's safe to ignore setting dataStoreVersionId.